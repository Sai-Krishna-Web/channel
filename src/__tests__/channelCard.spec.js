import React from 'react';
import { render } from '@testing-library/react';
import { shallow, mount } from 'enzyme';
import ChannelCard from '../components/channelCard';

const item={
description: "Nunc hendrerit blandit elit sed rhoncus. Sed interdum tempus enim vel ornare. Nulla facilisi. Morbi rhoncus turpis in justo sollicitudin, sit amet varius magna fringilla. Fusce porta magna neque, nec.",
instructorName: "Vestibulum Ante",
instructorPhotoUrl: "https://placeholdit.imgix.net/~text?txtsize=34&txt=B&w=60&h=60",
subjectPhotoUrl: "https://placeholdit.imgix.net/~text?txtsize=34&txt=A&w=60&h=60",
time: "2016-01-05 22:00:00",
title: "Ut consequat risus id lacus."
}
let wrapper=shallow(<ChannelCard item={item}/>);
describe('renders ChannelCard Component', () => {
    it('renders correctly', () => {
      wrapper =shallow(<ChannelCard item={item} />);
      expect(wrapper.exists()).toBe(true)
    });

    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
      });

    it('should match the product details content', () => {
        const component = mount(<ChannelCard item={item} />);
    
        expect(component.text()).toContain(item.title);
        expect(component.text()).toContain(item.description);
        expect(component.text()).toContain(item.instructorName);
      });
});

