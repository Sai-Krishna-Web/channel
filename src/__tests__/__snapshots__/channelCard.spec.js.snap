// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ChannelCard Component should match the snapshot 1`] = `ShallowWrapper {}`;
